<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CIoConfig</class>
 <widget class="QDialog" name="CIoConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>301</width>
    <height>371</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure I/O devices</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../icons.qrc">
    <normaloff>:/icons/icons/flash.svg</normaloff>:/icons/icons/flash.svg</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBoxInput">
     <property name="title">
      <string>I/Q input</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="inDevLabel1">
        <property name="text">
         <string>Device</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="inDevCombo">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Select an input device</string>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QComboBox::AdjustToMinimumContentsLength</enum>
        </property>
        <property name="minimumContentsLength">
         <number>15</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="inDevLabel2">
        <property name="toolTip">
         <string>Device arguments</string>
        </property>
        <property name="text">
         <string>Device string</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="inSrLabel">
        <property name="text">
         <string>Sample rate</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="inSrCombo">
        <property name="toolTip">
         <string>Select the sample rate of the input source</string>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="loLabel">
        <property name="toolTip">
         <string>LNB LO frequency. Use negative frequency for upconverters.</string>
        </property>
        <property name="text">
         <string>LNB LO</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="loSpinBox">
        <property name="toolTip">
         <string>LNB LO frequency. Use negative frequency for upconverters.</string>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::PlusMinus</enum>
        </property>
        <property name="suffix">
         <string> MHz</string>
        </property>
        <property name="decimals">
         <number>6</number>
        </property>
        <property name="minimum">
         <double>-15000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>15000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="inDevEdit">
        <property name="toolTip">
         <string>The device argument is a delimited string used to locate devices on your system.
Use the device id or name (if applicable) to specify a certain device or list
of devices. If left blank, the first device found will be used.

Examples (some arguments may be optional):
  fcd=0
  rtl=0,rtl_xtal=28.80001e6,tuner_xtal=26e6,buffers=64 ...
  rtl_tcp=127.0.0.1:1234,psize=16384
  uhd,subdev=A:0,label='USRP1'
  osmosdr=0|name,mcr=64e6,nchan=5,port=/dev/ttyUSB0 ...
  file=/path/to/file.ext,freq=428e6,rate=1e6,repeat=true,throttle=true ...

You can test the device strings in gnuradio-companion.</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="bwSpinBox">
        <property name="toolTip">
         <string>Analog bandwidth (leave at 0 for default)</string>
        </property>
        <property name="suffix">
         <string> MHz</string>
        </property>
        <property name="decimals">
         <number>6</number>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="bwLabel">
        <property name="toolTip">
         <string>Analog bandwidth (leave at 0 for default)</string>
        </property>
        <property name="text">
         <string>Bandwidth</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxOutput">
     <property name="title">
      <string>Audio output</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="outDevLabel">
        <property name="text">
         <string>Device</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="outDevCombo">
        <property name="toolTip">
         <string>Select which device to use for audio output.
Leave it at default unless you know what you are doing.</string>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QComboBox::AdjustToMinimumContentsLength</enum>
        </property>
        <property name="minimumContentsLength">
         <number>15</number>
        </property>
        <item>
         <property name="text">
          <string>Default</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="outSrLabel">
        <property name="text">
         <string>Sample rate</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="outSrCombo">
        <property name="toolTip">
         <string>Select the audio sample rate</string>
        </property>
        <item>
         <property name="text">
          <string>48 kHz</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CIoConfig</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CIoConfig</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
